tests/dot_type_call_ambiguity/dot_type_call_ambiguity.fz

```diff
@@ -0,0 +1,32 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion test
+#
+# -----------------------------------------------------------------------
+
+dot_type_call_ambiguity is
+
+  a => 3
+
+  type.a => 4
+
+  # NYI should result in an error: Duplicate feature declaration or similar.
```

Comment:
We can solve this later, but here are my thoughts: 

A `duplicate feature` is a bit difficult to define when you take visibility into account. 

I prefer to cause an error at an ambiguous call and to provide a means to disambiguate the calls:
```
ex is 
  a => 3
  type.a => 4

say ex.a    # should cause an error, ambiguous!
```
To make a call on the type feature, one could write
```
say ex.type.a
```
It is a bit harder to make a call on the instance, one ugly solution might be a temp field:
```
  temp := ex
  temp.a
```
Nicer might be to add an `instance` (or `self`, `me`, `this`) to `Any.fz`
```
public Any ref is
  ...
  public instance => Any.this
```
then we might be able to do
```
say ex.instance.a
```